### PR TITLE
Troubleshooting unit test failures

### PR DESCRIPTION
**Pull request by NeonDaniel**
Update logging to prevent Type Errors in testing
Add back `self.loaded` set in skill_loader.py
Add new bus handlers to test_skill_manager.py

### CODE REVIEW

File: test/unittests/skills/test_skill_manager.py [hunk 1]
```diff
@@ -111,10 +111,10 @@ def test_instantiate(self):
             'mycroft.skills.all_loaded',
             'mycroft.network.connected',
             'mycroft.internet.connected'
-        ]
-        self.assertListEqual(
+        }
+        self.assertSetEqual(
```
Comment (JarbasAl) [inline test/unittests/skills/test_skill_manager.py:115]: dont we want to test order also?

Comment (NeonDaniel) [reply test/unittests/skills/test_skill_manager.py:115]: Does order matter for bus events? I thought this was just checking which events are registered during init

Comment (JarbasAl) [reply test/unittests/skills/test_skill_manager.py:115]: I'd argue they matter, It reflects the order methods are called, but testing that was a side effect and its ok to deprecate.

Comment (NeonDaniel) [reply test/unittests/skills/test_skill_manager.py]: Fair enough, changed that back

**Pull request merged**